garak/generators/huggingface.py

```diff
@@ -106,13 +116,16 @@ def _call_model(
             warnings.simplefilter("ignore", category=UserWarning)
             try:
                 with torch.no_grad():
-                    # workaround for pipeline to truncate the input
-                    encoded_prompt = self.generator.tokenizer(prompt, truncation=True)
-                    truncated_prompt = self.generator.tokenizer.decode(
-                        encoded_prompt["input_ids"], skip_special_tokens=True
-                    )
+                    # according to docs https://huggingface.co/docs/transformers/main/en/chat_templating
+                    # chat template should be automatically utilized if the pipeline tokenizer has support
+                    # and a properly formatted list[dict] is supplied
+                    if self.use_chat:
+                        formatted_prompt = self._format_chat_prompt(prompt)
+                    else:
+                        formatted_prompt = prompt
```

Comment:
Note for future Erick and Jeffrey: we will want to merge `ConversationalPipeline` into this at some point given HF's deprecation of the `"conversational"` pipeline and `Conversation` object. At that point, we'll also want to validate that `prompt` is a string.